reactioncommerce/generator-reaction
# feat: add option for generating frontend project

## Changes

- You now run `yo reaction-next` instead of `yo reaction-next projectname`, and the first prompt is for the project name.
- Added `frontend` generator file, folder, template. There is a README in there explaining how I made that template starting from a `create-react-app` command.
- Moved some of `base` generator to a `backend` generator. The main `app` generator now composes with all other generators, but only frontend or backend will run because they do checks at the top of their methods to see what `projectType` was selected.
- The deletion of node_modules is now done outside of yo, allowing us to force it without prompting, and making it happen much faster due to no diffing.

NOTE: I disabled prettier for JSON files because I couldn't figure out how to make it ignore the package.json files in **/templates, which have invalid template tags. This was preventing commit due to precommit hooks.

## Testing

1. Build an app choosing `frontend` and then `docker-compose up` and make sure you see the dummy app in a browser at localhost:4000
1. Build an app choosing `backend` and then `docker-compose up` and make sure you see the graphql playground app in a browser at localhost:4000
